### ISSUE TITLE
Add example of matching catalogs with a separation constraint to docs

### ISSUE THREAD

**Issue by Gabriel-p** [astropy/astropy#5865]
I'm using the `match_to_catalog_sky()` function described in [Matching Catalogs](http://docs.astropy.org/en/stable/coordinates/matchsep.html#matching-catalogs).

For what I can see there's no way to impose a maximum distance separation to act as a match filter, something that could in principle be considered a basic necessity.

Is there a way to impose say a 5 sec maximum separation distance on this function? If not, could it be considered a future addition?

**Comment by adrn** [comment]
The catalog matching produces separations, which can then be used to impose a filter. For example:

```python
import astropy.units as u
import astropy.coordinates as coord
max_sep = 5 * u.arcsec
idx, sep, _ = coord.match_to_catalog_sky(coords1, coords2)
close_matches = idx[sep < max_sep]
```

I do think an example like this should be added to the documentation, but I don't think we should add an argument to the function for a maximum separation.

**Comment by bsipocz** [comment]
Addressed by #8341.
